src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -62,6 +69,13 @@ internal class RequestExecutionQueue<TRequestContext> : IRequestExecutionQueue<T
     protected readonly AsyncQueue<(IQueueItem<TRequestContext> queueItem, Guid ActivityId, CancellationToken cancellationToken)> _queue = new();
     private readonly CancellationTokenSource _cancelSource = new();
 
+    /// <summary>
+    /// Map of method to the handler info for each language.
+    /// The handler info is created lazily to avoid instantiating any types or handlers until a request is recieved for
+    /// that particular method and language.
+    /// </summary>
+    private readonly FrozenDictionary<string, FrozenDictionary<string, Lazy<(RequestHandlerMetadata Metadata, IMethodHandler Handler, MethodInfo MethodInfo)>>> _handlerInfoMap;
```

Comment:
Caching method infos for invoking handlers with specific parameters.

Reply:
nit: i recommend the Rewrap extension, set to 120 column.  Makes it easy to wrap comments so you don't have to mnually fix things up.